### PR TITLE
Propagate the -no-allocations frontend flag to the frontend

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
@swift-ci please test

File: Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift [hunk 1]
```diff
@@ -254,6 +254,7 @@ extension Driver {
     try commandLine.appendAll(.debugPrefixMap, .coveragePrefixMap, .filePrefixMap, from: &parsedOptions)
     try commandLine.appendAllArguments(.Xfrontend, from: &parsedOptions)
     try commandLine.appendLast(.warnConcurrency, from: &parsedOptions)
+    try commandLine.appendLast(.noAllocations, from: &parsedOptions)
```
Comment (artemcm) [inline Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift:257]: Could we please guard it with `if isFrontendArgSupported(.noAllocations)`

Comment (kubamracek) [reply Sources/SwiftDriver/Jobs/FrontendJobHelpers.swift]: Done!

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Comment by MaxDesiatov** [comment]
Apologies, pressed a wrong button when posting CI trigger and mistakenly closed it instead

**Comment by MaxDesiatov** [comment]
@swift-ci test windows

**Review by artemcm — APPROVED** [review]

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
@swift-ci please test Windows platform

**Pull request merged**